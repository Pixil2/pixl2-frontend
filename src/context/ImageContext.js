import { createContext, useEffect, useState } from 'react';
import { createImage } from '../utils/grid';

export const imageContext = createContext();

export function ProvideImage({ children }) {
  const [image, setImage] = useState({});
  const [eraser, setEraser] = useState({});
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    setImage(createImage('title', 15, 15));
    setEraser(createImage('eraser', 10, 10));
    setLoading(false);
  }, [loading]);

  return (
    <imageContext.Provider value={{ image, eraser, loading, setLoading }}>
      {children}
    </imageContext.Provider>
  );
}
