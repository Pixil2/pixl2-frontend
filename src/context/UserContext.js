import {
  createContext,
  useCallback,
  useContext,
  useEffect,
  useMemo,
  useState,
} from 'react';

import { signIn, signOut } from '../services/users';

const UserContext = createContext();

export const UserProvider = ({ children }) => {
  const [loading, setLoading] = useState(true);
  const [user, setUser] = useState(null);

  const login = async (credentials) => {
    try {
      const user = await signIn(credentials);
      setUser(user);
    } catch (error) {
      throw error;
    }
  };

  const logout = useCallback(() => {
    signOut().then(() => setUser(null));
  }, []);
};
