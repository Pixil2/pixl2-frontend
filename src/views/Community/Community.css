.Community {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.imageContainer {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: auto;
  gap: 2vw;
  padding-top: 6vw;
}

.profileGrid {
  border: 5px solid black;
}

.image {
  display: grid;
  grid-template-columns: repeat(10, 1fr);
  grid-template-rows: repeat(10, 1fr);
  height: 250px;
  width: 250px;
}

.imageHeader {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-bottom: 1vh;
}

.imageTitle {
  text-transform: capitalize;
  font-size: 2vw;
}

.imageTag {
  text-transform: lowercase;
  font-family: monospace;
}
.communityHeader {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 6vh 0 2vh 0;
  gap: 2vh;
}

.createButton {
  background-color: white;
  border: black 0.3vw solid;
  padding: 1vw;
  font-family: monospace;
  font-size: 2vw;
  margin-top: 2vh;
}
.createButton:hover {
  transform: scale(1.02);
}

.createButton:active {
  transform: scale(1.02);
  filter: drop-shadow(1px 1px 1px black);
}

.createButton {
  font-size: 2vw;
  margin-top: 2vh;
}
