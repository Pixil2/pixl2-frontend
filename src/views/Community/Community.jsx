import { useEffect, useState } from 'react';
import {
  getUserImages,
  deleteImageById,
  getAllImages,
} from '../../services/images';
import { getImageByTagId } from '../../services/tags';
import ProfileGrid from '../../components/Profile/ProfileGrid';
import { getCurrentUser } from '../../services/users';
import { v4 as uuid } from 'uuid';
import styles from './Community.css';
import { Link, useNavigate } from 'react-router-dom';
import Header from '../../components/Layout/Header';
import { getTagByImageId } from '../../services/tags';
import { useCurrentUser } from '../../context/UserContext';

export default function Community() {
  const [currentImages, setCurrentImages] = useState([]);
  const [animal, setAnimal] = useState([]);

  useEffect(() => {
    const fetch = async () => {
      const res = await getAllImages();
      const tagArray = [];
      await Promise.all(
        res.map(async (item) => {
          const tags = await getTagByImageId(item.id);
          tagArray.push({ ...tags });
        })
      );
      setCurrentImages(tagArray);
    };
    fetch();
  }, []);

  useEffect(() => {
    const fetch = async () => {
      const res = await getImageByTagId(1);
      console.log('res.images', res.images);
      setAnimal(res.images);
    };
    fetch();
  }, []);

  //   const getTaggedImages = async (id) => {
  //     const taggedImages = await getImageByTagId(id);
  //     console.log('taggedImages', taggedImages);
  //     return taggedImages;
  //   };

  //   console.log('animal', animal);

  return (
    <>
      <Header />
      <div className={styles.Community}>
        <div className={styles.communityHeader}>
          <h1 className={styles.communityTitle}>Community Gallery</h1>
          <p className={styles.communityCaption}>
            Hey, pixel people! Check out all of the awesome art that has been
            created by our community.
          </p>
        </div>
        <Link to="/canvas">
          <button className={styles.createButton}>CREATE IMAGE</button>
        </Link>
        <div className={styles.imageContainer}>
          {currentImages.map((item, index) => {
            const tag = item.tags[0].name;
            console.log(tag);
            return (
              <div key={uuid()}>
                <div className={styles.imageHeader}>
                  <h1 className={styles.imageTitle}>{item.title}</h1>
                  <p className={styles.imageTag}>TAG: {tag}</p>
                </div>
                <ProfileGrid image={item} />
              </div>
            );
          })}
        </div>
        <div className={styles.animalContainer}>
          {/* {animal.map((item, index) => {
          const tag = item.tags.name;
          console.log(tag);
          return (
            <div key={uuid()}>
              <div className={styles.imageHeader}>
                <h1 className={styles.imageTitle}>{item.title}</h1>
                <p className={styles.imageTag}>TAG: {tag}</p>
              </div>
              <ProfileGrid image={item} />
            </div>
          );
        })} */}
        </div>
      </div>
    </>
  );
}
