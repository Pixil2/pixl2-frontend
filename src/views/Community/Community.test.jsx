import {
  render,
  screen,
  waitForElementToBeRemoved,
} from '@testing-library/react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { MemoryRouter, Routes, Route } from 'react-router-dom';
import { UserProvider } from '../../context/UserContext';
import Community from './Community';

const image = {
  id: '111111',
  title: 'A title',
  height: 10,
  width: 10,
  colorArray: ['rgb(0, 0, 0)'],
  userId: '1',
  isPublic: false,
  isApproved: null,
};

const user = {
  id: 100000000000,
  username: 'testUser',
};

const server = setupServer(
  rest.get(`${process.env.API_URL}/api/v1/images`, (req, res, ctx) => {
    return res(ctx.json([image]));
  }),
  rest.get(`${process.env.API_URL}/api/v1/users/me`, (req, res, ctx) => {
    return res(ctx.json(user));
  }),
  rest.post(`${process.env.API_URL}/api/v1/users/me`, (req, res, ctx) => {
    return res(ctx.json(user));
  }),
  rest.get(`${process.env.API_URL}/api/v1/tags/1/images`, (req, res, ctx) => {
    return res(ctx.json([image]));
  })
);

describe('Community', () => {
  beforeAll(() => {
    server.listen();
  });

  afterAll(() => {
    server.close();
  });

  it.skip('Tests functionality of Community view', async () => {
    const { container } = render(
      <UserProvider>
        <MemoryRouter initialEntries={['/community']}>
          <Routes>
            <Route
              exact
              path="/community"
              element={
                <Community
                  user={{
                    id: 1,
                    username: 'whatever',
                  }}
                  image={{ image }}
                />
              }
            />
          </Routes>
        </MemoryRouter>
      </UserProvider>
    );
    const loading = screen.getByText(/loading/i);
    expect(loading).toBeInTheDocument();
    await waitForElementToBeRemoved(() => screen.getByText(/loading/i));

    const createButton = await screen.findByRole('button', {
      name: /create image/i,
    });
    expect(createButton).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
