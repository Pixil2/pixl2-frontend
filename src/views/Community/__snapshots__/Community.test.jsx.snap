// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Community Tests functionality of Community view 1`] = `
<div>
  <div
    class="communityContainer"
  >
    <header
      class="Header"
    >
      <img
        class="headerLogo"
        src="/path/image"
      />
      <div
        class="headerLinkContainer"
      >
        <a
          class="headerLink"
          href="/profile"
        >
          Profile
        </a>
        <a
          class="headerLink"
          href="/about"
        >
          About
        </a>
        <button
          class="headerButton"
        >
          SIGN OUT
        </button>
      </div>
    </header>
    <div
      class="communityContainer"
    >
      <div
        class="communityHeader"
      >
        <h1
          class="communityTitle"
        >
          Community Gallery
        </h1>
        <p
          class="communityCaption"
        >
          Hey, pixel people! Check out all of the awesome art that has been created by our community.
        </p>
      </div>
      <a
        href="/canvas"
      >
        <button
          class="createButton"
        >
          CREATE IMAGE
        </button>
      </a>
      <div
        class="imageContainer"
      />
    </div>
  </div>
</div>
`;
